sphinx/ext/inheritance_diagram.py

```diff
@@ -184,13 +185,11 @@ def recurse(cls: Any) -> None:
 
             # Use first line of docstring as tooltip, if available
             tooltip = None
-            try:
+            with suppress(AttributeError):  # might raise AttributeError for strange classes
                 if cls.__doc__:
                     doc = cls.__doc__.strip().split("\n")[0]
                     if doc:
                         tooltip = '"%s"' % doc.replace('"', '\\"')
-            except Exception:  # might raise AttributeError for strange classes
```

Comment:
Note: The contradiction between implementation and comment started from the beginning (3f7ecbde96). I don't know which is correct. But let's go with AttributeError.